toolbox/util/githubClient.go

```diff
@@ -284,26 +284,69 @@ func (g GithubClient) GetTagCommitSHA(repo, tag string) (string, error) {
 }
 
 // GetCommitCreationTime gets the time when the commit identified by sha is created
-func (g GithubClient) GetCommitCreationTime(repo, sha string) (*time.Time, error) {
+func (g GithubClient) GetCommitCreationTime(repo, sha string) (time.Time, error) {
 	commit, _, err := g.client.Git.GetCommit(
 		context.Background(), g.owner, repo, sha)
 	if err != nil {
 		log.Printf("Failed to get commit %s", sha)
-		return nil, err
+		return time.Time{}, err
 	}
-	return (*(*commit).Author).Date, nil
+	return commit.Author.GetDate(), nil
 }
 
 // GetCommitCreationTimeByTag finds the time when the commit pointed by a tag is created
 // Note that SHA of the tag is different from the commit SHA
-func (g GithubClient) GetCommitCreationTimeByTag(repo, tag string) (*time.Time, error) {
+func (g GithubClient) GetCommitCreationTimeByTag(repo, tag string) (time.Time, error) {
 	commitSHA, err := g.GetTagCommitSHA(repo, tag)
 	if err != nil {
-		return nil, err
+		return time.Time{}, err
 	}
 	return g.GetCommitCreationTime(repo, commitSHA)
 }
 
+// GetReleaseCreationTime gets the creation time of a release tag (0.1.6, 0.2.7)
+func (g GithubClient) GetReleaseCreationTime(repo, tag string) (createTime time.Time, err error) {
+	if repo == "istio" {
+		createTime, err = g.getReleaseTagCreationTime(repo, tag)
+	} else {
+		createTime, err = g.getannotatedTagCreationTime(repo, tag)
+	}
+	if err != nil {
+		log.Printf("Cannot get the creation time of %s/%s", repo, tag)
+		return time.Time{}, err
+	}
+	return createTime, nil
+}
+
+// getReleaseTagCreationTime gets the creation time of a lightweight tag created by release
+// Release tags from istio/istio are this kind of tag.
+func (g GithubClient) getReleaseTagCreationTime(repo, tag string) (time.Time, error) {
+	release, _, err := g.client.Repositories.GetReleaseByTag(context.Background(), g.owner, repo, tag)
+	if err != nil {
+		return time.Time{}, fmt.Errorf("failed to to get release tag: %s", err)
+	}
+	return release.GetCreatedAt().Time, nil
+}
+
+// getannotatedTagCreationTime gets the creation time of an annotated Tag
+// Release tags except those from istio/istio are annotated tag.
+func (g GithubClient) getannotatedTagCreationTime(repo, tag string) (time.Time, error) {
+	sha, ty, err := g.GetReference(repo, "refs/tags/"+tag)
```

Comment:
This is tangent to this PR but it would be nice to rename `GetReference` to `GetReferenceSHAAndType` (and the function comments) so there is no ambiguous in what it returns (ref oibject vs. sha and ty)